packages/mdc-textfield/adapter.js

```diff
@@ -196,6 +175,12 @@ class MDCTextFieldAdapter {
    * @return {?MDCTextFieldHelperTextFoundation}
    */
   getHelperTextFoundation() {}
+
+  /**
+   * Returns the foundation for the input element.
+   * @return {?MDCTextFieldInputFoundation}
```

Comment:
I think the input foundation is actually required (unlike helper text, which was optional). So this should be `@return {MDCTextFieldInputFoundation}` (no ?)

Reply:
Usually in cases like this, we should have `!` instead of `?`, right? I mean, it shouldn't be null either?

Reply:
Yeah, `{MDCTextFieldInputFoundation}` implies `{?MDCTextFieldInputFoundation}`, so I changed the `?` to `!`